Upstream-Status: Backport

Backported from https://quiche.googlesource.com/quiche/+/4f43a02b2a83a0d836f8fdd6863bb9523f9c1748

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From c4da3cf60dcf6854804ed6652665497da6ca8b63 Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Wed, 10 Mar 2021 11:12:22 +0200
Subject: [PATCH] Fix use of DCHECK with std::unique_ptr

DCHECK_NE comparison requires CheckOpValueStr to be defined for the
type, or providing an output stream operator. A unique_ptr does not
provide any. USE DCHECK instead.

This change was originally written by Stephan Hartmann <https://quiche-review.googlesource.com/c/quiche/+/10680>.

PiperOrigin-RevId: 353340281
Change-Id: I0bf24145007ba7e7ae2bc8a04301ec9a300e335c
---
 net/third_party/quiche/src/quic/core/quic_path_validator.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/third_party/quiche/src/quic/core/quic_path_validator.cc b/net/third_party/quiche/src/quic/core/quic_path_validator.cc
index ee446cc2..0cc387f6 100644
--- a/net/third_party/quiche/src/quic/core/quic_path_validator.cc
+++ b/net/third_party/quiche/src/quic/core/quic_path_validator.cc
@@ -68,7 +68,7 @@ void QuicPathValidator::OnPathResponse(const QuicPathFrameBuffer& probing_data,
 void QuicPathValidator::StartPathValidation(
     std::unique_ptr<QuicPathValidationContext> context,
     std::unique_ptr<ResultDelegate> result_delegate) {
-  DCHECK_NE(nullptr, context);
+  DCHECK(context);
   QUIC_DLOG(INFO) << "Start validating path " << *context
                   << " via writer: " << context->WriterToUse();
   if (path_context_ != nullptr) {
-- 
2.25.1

